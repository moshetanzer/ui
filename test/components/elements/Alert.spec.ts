import { describe, it, expect } from 'vitest'
import { UAlert } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Alert', () => {
  it.each([
    ['with title', { props: { title: 'Alert Title' } }],
    ['with icon', { props: { title: 'Alert Title', icon: 'heroicons-information-circle' } }],
    ['with avatar', { props: { title: 'Alert Title', avatar: { src: 'avatar.jpg', alt: 'User Avatar' } } }],
    ['with close button', { props: { title: 'Alert Title', closeButton: { label: 'Close' } } }],
    ['with actions', { props: { title: 'Alert Title', actions: [() => { }] } }],
    ['with custom color', { props: { title: 'Alert Title', color: 'red' } }],
    ['with custom variant', { props: { title: 'Alert Title', variant: 'outline' } }],
    ['with title slot', { slots: { title: () => 'Title slot' } }],
    ['with description slot', { slots: { description: () => 'Description slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UAlert.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UAlert)
    expect(html).toMatchSnapshot()
  })
})
