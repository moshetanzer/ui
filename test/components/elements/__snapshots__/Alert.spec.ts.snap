// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > renders with actions correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <div class="flex items-center gap-2 flex-shrink-0 mt-0"><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-xs gap-x-1.5 p-1.5 text-primary-500 hover:text-primary-600 disabled:text-primary-500 dark:text-primary-400 dark:hover:text-primary-500 dark:disabled:text-primary-400 underline-offset-4 hover:underline focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center"></button>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`Alert > renders with avatar correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center">
    <!--v-if--><span class="relative inline-flex items-center justify-center flex-shrink-0 rounded-full h-10 w-10 text-base flex-shrink-0 self-center"><img class="rounded-full h-10 w-10 text-base" alt="User Avatar" src="avatar.jpg"><!--v-if--></span>
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with close button correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <div class="flex items-center gap-2 flex-shrink-0 mt-0">
      <!--v-if--><button type="button" class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" aria-label="Close">
        <!--v-if--><span class="">Close</span>
        <!--v-if-->
      </button>
    </div>
  </div>
</div>"
`;

exports[`Alert > renders with custom color correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 bg-red-500 dark:bg-red-400 text-white dark:text-gray-900">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with custom variant correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-white-500 dark:text-white-400 ring-1 ring-inset ring-white-500 dark:ring-white-400">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with description slot correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-start">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <!--v-if-->
      <p class="mt-1 text-sm leading-4 opacity-90">Description slot</p>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with icon correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center"><span class="heroicons-information-circle flex-shrink-0 w-5 h-5"></span>
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with title correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with title slot correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex gap-3 items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="w-0 flex-1">
      <p class="text-sm font-medium">Title slot</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;
